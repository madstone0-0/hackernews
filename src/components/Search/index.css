.search .form > button.btn.btn-primary {
  margin-bottom: 5px;
}

.search .search-bar:focus {
  border: 1px solid #0069D9;
}

@media only screen and (max-width: 768px) {
  .search .search-bar {
    width: 270px;
    margin-right: 5px;
  }
  .search .search-button {
    margin-bottom: 8px;
  }
}

@media only screen and (max-width: 375px) {
  .search .search-button {
    margin-top: 5px;
  }
}
