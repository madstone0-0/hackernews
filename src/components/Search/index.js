// eslint-disable-next-line no-unused-vars
import React, { Component } from "react";
import PropTypes from "prop-types";

// Search Component
class Search extends Component {
    render() {
        const { value, onChange, onSubmit, children } = this.props;
        return (
            <form onSubmit={onSubmit}>
                <input
                    type="text"
                    value={value}
                    onChange={onChange}
                />
                <button type="submit" className="btn btn-info">
                    <span className="glyphicon glyphicon-search"/> {children}
                </button>
            </form>
        );
    }
}

Search.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
    onSubmit: PropTypes.func.isRequired,
};

export default Search;