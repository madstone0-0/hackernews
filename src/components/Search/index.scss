.search {
    .form > button.btn.btn-primary {
        margin-bottom: 5px;
    }

    @media only screen and (max-width: 768px) {
        .search-bar {
            width : 270px;
            margin-right: 5px;
        }

        .search-button {
            
            margin-bottom: 8px;
        }
    }

    @media only screen and (max-width: 375px) {
        .search-button {
            margin-top: 5px;
        }
    }
}