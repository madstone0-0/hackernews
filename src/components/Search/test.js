/* eslint-disable no-unused-vars,no-undef */
/* eslint-disable no-console */
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import React from "react";
import Search from "./index";

describe("Search", function() {
    it("renders", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Search onChange={() => {
            console.log("Change");
        }} onSubmit={() => {
            console.log("Submitted");
        }}>Search</Search>, div);
    });

});

test("snapshots", () => {
    const component = renderer.create(
        <Search onChange={() => {
            console.log("Change");
        }} onSubmit={() => {
            console.log("Submitted");
        }}> Search </Search>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});