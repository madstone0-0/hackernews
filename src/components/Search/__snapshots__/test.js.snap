// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    className="btn btn-info"
    type="submit"
  >
    <span
      className="glyphicon glyphicon-search"
    />
     
     Search 
  </button>
</form>
`;
